proto/pulumi/resource.proto

```diff
@@ -170,15 +170,22 @@ message RegisterResourceRequest {
     string packageRef = 33; // a reference from RegisterPackageRequest.
 
     message ResourceHooksBinding {
+        // Lifecycle hooks
         repeated string before_create = 1;
         repeated string after_create = 2;
         repeated string before_update = 3;
         repeated string after_update = 4;
         repeated string before_delete = 5;
         repeated string after_delete = 6;
+        
+        // Error hooks (single hook per type)
+        optional string on_error_create = 7;
```

Comment:
I wonder if we should merge all these together and send the operation as a parameter instead? Feels likely that users will want to retry all operations a lot of the time and not want to setup three hooks for that.